cmd/main.go

```diff
@@ -75,65 +96,76 @@ func init() {
-		probeAddr                      string
-
-		restConfigQPS                        int
-		restConfigBurst                      int
-		linodeClusterConcurrency             int
-		linodeMachineConcurrency             int
-		linodeObjectStorageBucketConcurrency int
-		linodeVPCConcurrency                 int
-		linodePlacementGroupConcurrency      int
-		linodeFirewallConcurrency            int
-	)
-	flag.StringVar(&machineWatchFilter, "machine-watch-filter", "", "The machines to watch by label.")
-	flag.StringVar(&clusterWatchFilter, "cluster-watch-filter", "", "The clusters to watch by label.")
-	flag.StringVar(&objectStorageBucketWatchFilter, "object-storage-bucket-watch-filter", "", "The object bucket storages to watch by label.")
-	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
-	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
-	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
-		"Enable leader election for controller manager. "+
-			"Enabling this will ensure there is only one active controller manager.")
-	flag.IntVar(&restConfigQPS, "kube-api-qps", qpsDefault,
-		"Maximum queries per second from the controller client to the Kubernetes API server. Defaults to 20")
-	flag.IntVar(&restConfigBurst, "kube-api-burst", burstDefault,
-		"Maximum number of queries that should be allowed in one burst from the controller client to the Kubernetes API server. Default 30")
-	flag.IntVar(&linodeClusterConcurrency, "linodecluster-concurrency", concurrencyDefault,
-		"Number of LinodeClusters to process simultaneously. Default 10")
-	flag.IntVar(&linodeMachineConcurrency, "linodemachine-concurrency", concurrencyDefault,
-		"Number of LinodeMachines to process simultaneously. Default 10")
-	flag.IntVar(&linodeObjectStorageBucketConcurrency, "linodeobjectstoragebucket-concurrency", concurrencyDefault,
-		"Number of linodeObjectStorageBuckets to process simultaneously. Default 10")
-	flag.IntVar(&linodeVPCConcurrency, "linodevpc-concurrency", concurrencyDefault,
-		"Number of LinodeVPCs to process simultaneously. Default 10")
-	flag.IntVar(&linodePlacementGroupConcurrency, "linodeplacementgroup-concurrency", concurrencyDefault,
-		"Number of Linode Placement Groups to process simultaneously. Default 10")
-	flag.IntVar(&linodeFirewallConcurrency, "linodefirewall-concurrency", concurrencyDefault,
-		"Number of Linode Firewall to process simultaneously. Default 10")
-	opts := zap.Options{
-		Development: true,
+	flags, opts := parseFlags()
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	setupLog.Info(fmt.Sprintf("CAPL version: %s", version.GetVersion()))
+
+	linodeClientConfig, dnsClientConfig := validateEnvironment()
+	mgr := setupManager(flags, linodeClientConfig, dnsClientConfig)
+
+	ctx := ctrl.SetupSignalHandler()
+	run := func(ctx context.Context) error {
+		o11yShutdown := setupObservabillity(ctx)
```

Comment:
Nitpick: fix spelling for `setupObservabillity` to `setupObservability`

Reply:
oops good catch!